### PR TITLE
Add log arg to gateway

### PR DESCRIPTION
**Pull request by pimeys**
Closes: GB-6344

```
      --log <LOG_LEVEL>
          Set the logging level
          
          [env: GRAFBASE_LOG=]

          Possible values:
          - off:   Completely disables logging
          - error: Only errors from Grafbase libraries
          - warn:  Warnings and errors from Grafbase libraries
          - info:  Info, warning and error messages from Grafbase libraries
          - debug: Debug, info, warning and error messages from all dependencies
          - trace: Trace, debug, info, warning and error messages from all dependencies

      --log-style <LOG_STYLE>
          Set the style of log output
          
          [env: GRAFBASE_LOG_STYLE=]
          [default: text]

          Possible values:
          - text: Standard text
          - json: JSON objects
```

### CODE REVIEW

File: gateway/crates/federated-server/src/server.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mod bind;
+// mod bind;
```
Comment (tomhoule) [inline gateway/crates/federated-server/src/server.rs:1]: The file is deleted so I guess this line can go.

File: gateway/crates/gateway-binary/src/args.rs [hunk 2]
```diff
@@ -43,9 +121,12 @@ pub struct Args {
...
     #[arg(long, short, env = "GRAFBASE_SCHEMA_PATH")]
     pub schema: Option<PathBuf>,
-    /// Set the tracing level
-    #[arg(short, long, default_value_t = 0)]
-    pub trace: u16,
+    /// Set the logging level
+    #[arg(long = "log", env = "GRAFBASE_LOG")]
+    pub log_level: Option<LogLevel>,
+    /// Set the style of log output
+    #[arg(long, env = "GRAFBASE_LOG_STYLE", default_value_t = LogStyle::Ascii)]
```
Comment (tomhoule) [inline gateway/crates/gateway-binary/src/args.rs:128]: Maybe we could default to LogStyle::Ansi when the terminal is interactive? I don't mind either way though, just a suggestion.

**Review by tomhoule — APPROVED** [review]
neat!

File: gateway/crates/gateway-binary/src/args.rs (hunk 2 above)
Comment (pimeys) [reply gateway/crates/gateway-binary/src/args.rs:128]: great idea!

**Pull request merged**